examples/rust/Cargo.toml

```
@@ -29,20 +29,14 @@ unwrap_used = "warn"
 wildcard_imports = "warn"
 
 [dependencies]
-# To allow `cargo-generate` to work, explicitly define dependency versions rather than importing the repo's workspace `Cargo.toml`.
-aranya-client = { version = "0.5.1" }
-aranya-daemon = { version = "0.5.1" }
-aranya-daemon-api = { version = "0.5.1" }
-aranya-util = { version = "0.5.1" }
+aranya-client = { git = "https://github.com/aranya-project/aranya.git" }
```

If we use a path here, we can ensure the example/ci job is running on the current checked out branch of the repo rather than `main`.
```suggestion
aranya-client = { path = "../../crates/aranya-client" }
```
@jdygert-spok any better ideas for handling this?

---

My concern is this may not work when a user is trying to generate a template from the repo without checking out the repo first.

---

The path approach won't work with the cargo-generate command. It will create the new project, but still look for a relative path which wouldn't exist. 

What we may want to do is loose the cargo generate approach all together and just focus on using the aranya-example. Then, we can use relative paths and be able to test the current state of the codebase in CI.

---

I agree that the Rust example is more important than cargo generate working.
If we have a running example, it should be simple enough for a user to copy the example to another project.

Another idea: copy a different `Cargo.toml` file into the template directory when running the example than what's used for the template by default. This would allow `cargo generate` to work as expected and give us more control over what branch is used by `examples/rust` and ci.
If we do this, I'd want to copy the entire template to a temporary directory first that is `.gitignore`'ed so we don't end up checking in the alternate `Cargo.toml` to the template.

---

We should check with @mooreoak and @jdygert-spok first before removing the template since `cargo generate` is intended to make it easier for a user to setup an example repo.